ivy/functional/ivy/device.py

```diff
@@ -378,6 +378,39 @@ def as_native_dev(device: Union[ivy.Device, ivy.NativeDevice], /) -> ivy.NativeD
     return ivy.current_backend().as_native_dev(device)
 
 
+@handle_exceptions
+def is_native_dev(device, /) -> bool:
+    """
+    Determine whether the input device is a Native device.
+
+    Parameters
+    ----------
+    dtype_in
+        The input device to check if it's a native device.
+        Determine whether the input data type is a native data type object.
+
+    Returns
+    -------
+    ret
+        Boolean, whether or not dtype_in is a native device.
+
+    Examples
+    --------
+    >>> ivy.set_backend("tensorflow")
+    >>> ivy.is_native_dev('/TPU:3')
+    True
+
+    >>> import torch
+    >>> device = torch.device("cuda")
+    >>> device
+    device(type='cuda')
+
+    >>> ivy.is_native_dev(device)
+    True
+    """
+    return ivy.current_backend().is_native_device(device)
+
```

Comment:
Also, the tests for the device-related functions you modified might have started failing now. You will probably need to update the assertions there to reflect the new expected behavior.